### PR TITLE
Food processors and microwaves now respect food trays

### LINKED ISSUE #53646: Vend-a-trays? Vendor trays? I just want to sell food, for god's sake!
[Round ID]: Literally any round you trontobringus

## Testmerges: 
I remember now why I hate making issue reports
## Reproduction:
1) Find a vending tray
2) Swipe your ID on it (thud)
3) Try to put a brainburger (your mother's recipe) or any food on it
4) The tray (display case) tells you to fuck off with that shit 

I should also note that you can't put food in the food cart either, making almost all of my food-based RP extremely difficult

### LINKED ISSUE #54812: Food trays are fuck and dont work properly
## Round ID:
///////

## Testmerges:
///////

## Reproduction:
Buy one bug get one free -

Issue 1 -
If you buy a tray from a vendor, the button at the top left (where the PDA light and spells are) will not appear due to the tray instantly appearing in your hand.
To fix it, you must pick drop it then pick it up.

Issue 2 -
You cannot insert the refactored food into microwaves or food processors.
For example, pick up four cutlets (only four cos cutlets are normal sized?) with the tray.
Click on the food processor with the tray.
Watch as absolutely bloody nothing happens when it is meant to deposit the four cutlets into the food processor.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Really it's a bandaid as it would be better to wait until the refactor is done, but it turned out to be a rather easy fix.
Food trays may now once again mass insert both new and old food into the microwaves and food processors.
Adds a static typecache between the two machines to compare against when handling mass-insertion, and handles the basic sorting/denying.

## Why It's Good For The Game

Fixes #53646 and Fixes #54812.
Prevents any manual handing when cooking large quantities of food at once. 
Also, you get the switch gathering mode button when being given the serving tray again.

## Changelog
:cl:
fix: Food trays may deposit old and new foods into the microwave and food processor again.
fix: Food trays give you the switch gathering mode button when bought from a vending machine again.
/:cl:

### CODE REVIEW

**Comment by brotherbeyondd** [comment]
MY SAVIOUR

**Review by ghost — APPROVED** [review]

**Comment by Qustinnus** [comment]
seems vend-a-tray's also check for the old food path directly, might be good to make a display_case proc for "item_can_be_inserted" or something because right now its made to just check for a specific type (cringe)

**Review by ExcessiveUseOfCobblestone — DISMISSED** [review]
why is this not IS_EDIBLE?

**Comment by ExcessiveUseOfCobblestone** [comment]
Why not use IS_EDIBILE?

**Comment by Qustinnus** [comment]
wew good point, I don't know why I approved a typecheck for obj/item/food. Try to use IS_EDIBLE() Instead.

**Comment by ArcaneMusic** [comment]
Oh, that got merged?
Great, on it.

**Comment by ArcaneMusic** [comment]
Done.

File: code/modules/food_and_drinks/kitchen_machinery/processor.dm [hunk 1]
```diff
@@ -14,6 +14,9 @@
 	var/processing = FALSE
 	var/rating_speed = 1
 	var/rating_amount = 1
+	var/static/list/allowed_cookables = typecacheof(list(
+		/obj/item/food,
+		/obj/item/reagent_containers/food))
```
Comment (LemonInTheDark) [inline code/modules/food_and_drinks/kitchen_machinery/processor.dm:19]: Unused, bad 🗞️

File: code/game/objects/structures/displaycase.dm [hunk 2]
```diff
@@ -577,3 +579,9 @@
 /obj/structure/displaycase/forsale/kitchen
 	desc = "A display case with an ID-card swiper. Use your ID to purchase the contents. Meant for the bartender and chef."
 	req_one_access = list(ACCESS_KITCHEN, ACCESS_BAR)
+
+/obj/structure/displaycase/forsale/insert_showpiece(obj/item/wack, mob/user)
+	if(!IS_EDIBLE(wack))
+		to_chat(user, "<span class='notice'>\The [src] smartly rejects [wack], as it only accepts food and drink.</span>")
```
Comment (Qustinnus) [inline code/game/objects/structures/displaycase.dm:585]:
```suggestion
		to_chat(user, "<span class='notice'>\The [src] smartly rejects [wack], as it only accepts food and drinks.</span>")
```

**Pull request merged**